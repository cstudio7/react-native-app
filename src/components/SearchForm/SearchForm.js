import React from 'react';
import { Animated, View, StyleSheet } from 'react-native';
import { Item, Input, Icon } from 'native-base';
import { Spacing } from '../../constants';

class SearchForm extends React.PureComponent {
  render() {
    return (
      <Animated.View
        style={[
          styles.searchForm,
          {
            transform: [
              {
                translateY: this.props.headerY
              }
            ]
          }
        ]}>
        <Item rounded>
          <Icon active name="md-search" />
          <Input
            placeholder="Поиск"
            value={this.props.target}
            onChangeText={this.props.changeTextHandler}
          />
          {this.props.target ? (
            <Icon
              active
              name="md-close"
              onPress={this.props.clearSearchFormHandler}
            />
          ) : null}
        </Item>
      </Animated.View>
    );
  }
}

const styles = StyleSheet.create({
  searchForm: {
    paddingTop: Spacing.base,
    paddingLeft: Spacing.base,
    paddingRight: Spacing.base,
    backgroundColor: 'white',
    paddingBottom: Spacing.base,
    backgroundColor: 'white',
    width: '100%',
    position: 'absolute',
    elevation: 0,
    flex: 1,
    zIndex: 1
  }
});

export default SearchForm;
