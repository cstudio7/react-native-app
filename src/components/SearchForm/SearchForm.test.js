import React from 'react';
import renderer from 'react-test-renderer';
import { Animated } from 'react-native';
import { Input } from 'native-base';
import SearchForm from './SearchForm';

const getInstance = props => {
  return renderer.create(<SearchForm {...props} />).root;
};

const getTree = props => {
  return renderer.create(<SearchForm {...props} />);
};

it('renders search form correctly', () => {
  const scrollY = new Animated.Value(0);
  const headerY = Animated.multiply(Animated.diffClamp(scrollY, 0, 72), -1);

  const props = {
    headerY,
    target: '',
  };

  const tree = getTree(props);
  expect(tree).toMatchSnapshot();
});

it('triggers onChangeText when the user typed out something', () => {
  const onChangeTextHandler = jest.fn();

  const props = {
    target: '',
    onChangeTextHandler
  };

  const instance = getInstance(props);
  instance.findByType(Input).props.onChangeText();
  expect(props.onChangeTextHandler).toHaveBeenCalled();
});

it('triggers onPressHandler when the user clean up the form', () => {
  const onChangeTextHandler = jest.fn();
  const onPressHandler = jest.fn();

  const props = {
    target: '123',
    onChangeTextHandler,
    onPressHandler
  };

  const instance = getInstance(props);
  instance.findByProps({ name: 'md-close' }).props.onPress();
  expect(onPressHandler).toHaveBeenCalled();
});
