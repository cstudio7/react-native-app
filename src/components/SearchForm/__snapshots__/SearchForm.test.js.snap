// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders search form correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "elevation": 0,
      "flex": 1,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
      "position": "absolute",
      "transform": Array [
        Object {
          "translateY": -0,
        },
      ],
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": 1,
        "borderColor": "#D9D5DC",
        "borderLeftWidth": 1,
        "borderRadius": 30,
        "borderRightWidth": 1,
        "borderTopWidth": 1,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginLeft": 2,
        "opacity": 1,
      }
    }
  >
    <Text
      active={true}
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#000",
              "fontSize": 30,
            },
            Object {
              "color": "#000",
              "fontSize": 24,
              "paddingLeft": 10,
              "paddingRight": 8,
            },
          ],
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <TextInput
      allowFontScaling={true}
      editable={true}
      placeholder="Поиск"
      placeholderTextColor="#575757"
      style={
        Object {
          "color": "#000",
          "flex": 1,
          "fontSize": 17,
          "height": 50,
          "paddingLeft": 8,
          "paddingRight": 5,
          "top": 1.5,
        }
      }
      underlineColorAndroid="rgba(0,0,0,0)"
      value=""
    />
  </View>
</View>
`;
